python-frontend/src/main/java/org/sonar/python/types/v2/TypeCheckBuilder.java

```diff
@@ -125,4 +136,15 @@ public TriBool test(PythonType pythonType) {
       return pythonType.equals(expectedType) ? TriBool.TRUE : TriBool.FALSE;
     }
   }
+
+  record IsInstanceOfPredicate(PythonType pythonType)  implements TypePredicate {
+
+    @Override
+    public TriBool test(PythonType pythonType) {
+      if (pythonType instanceof ObjectType objectType) {
+        return this.pythonType.equals(objectType.unwrappedType()) ? TriBool.TRUE : TriBool.FALSE;
+      }
+      return TriBool.UNKNOWN;
+    }
+  }
```

Comment:
As stated in my previous comment, this is not correct.

```
class Parent: ...
class Child(Parent): ...
c = Child()
isinstance(c, Parent) # True
```

This predicate, based on type equality, would incorrectly return false in such a case.

I understand you want to test that the type is an actual instance (`ObjectType`) and not something like a `ClassType`; however, the method naming is misleading. If you really need to make that test, you may combine a predicate that ensures a type is a `ObjectType` with the one for `IsSameTypeAs` (which currently simply unwraps `ObjectType` so that it would return true for an object or for the type itself).

Note that, `ClassType` being an instance of `type` is also an edge case to take into account (simply checking `ObjectType` vs `ClassType` vs `FunctionType` may not be enough depending on context, since everything is ultimately an object in Python)